packages/react-server-cli/src/coreJsMiddleware.js

```diff
@@ -1,17 +1,33 @@
-export default (pathToStatic) => {
+export default (pathToStatic, manifest) => {
 	return class CoreJsMiddleware {
 		getSystemScripts(next) {
 			const routeName = this.getRequest().getRouteName();
 			const baseUrl = pathToStatic || "/";
+
+			const scripts = [];
+
+			// read out the chunkmap so that the client code knows how to download the
+			// other chunks.
+			scripts.push({
+				type: "text/javascript",
+				text: `window.webpackManifest = ${JSON.stringify(manifest.jsChunksById)};`,
+			});
+
+			scripts.push(`${baseUrl}${manifest.jsChunksByName.common}`);
+			scripts.push(`${baseUrl}${manifest.jsChunksByName[routeName]}`);
+
 			return [
-				`${baseUrl}common.js`,
-				`${baseUrl}${routeName}.bundle.js`,
+				...scripts,
 				{
 					type: "text/javascript",
-					text: baseUrl ? `
-						if (typeof window !== "undefined" && window.__setReactServerBase) {
-							window.__setReactServerBase(${JSON.stringify(baseUrl)});
-						}` : "",
+					text: `
+						if (typeof window !== "undefined") {
+							if (window.__setReactServerBase) {
+								window.__setReactServerBase(${JSON.stringify(baseUrl)});
+							}
+							window.__reactServerManifest = ${manifest ? JSON.stringify(manifest) : "{}"};
```

Comment:
This could get big for a large site, right?  And it is, itself, static for a build?  I wonder if there's a way we could sneak it into the common bundle...